### PR TITLE
RHDEVDOCS-3009 Add info on Kubernetes matchLabels to "Forwarding application logs from specific pods" topic

### PR DESCRIPTION
**Pull request by rolfedh**
- Aligned team: Dev Tools
- For branches: master, enterprise-4.7, enterprise-4.8
- https://issues.redhat.com/browse/RHDEVDOCS-3009
- Direct link to doc preview: Not available. This topic is temporarily unpublished (not included) in master and enterprise-4.7. It will be re-published at the end of May, when OpenShift Logging 5.1 is GA. For now, please review the file as shown in the "Files changed" tab, above.
- SME review: Reviewed by @sichvoge  
- QE review: Reviewed by @kabirbhartiRH 
- Peer review: Reviewed by @jc-berger and @Preeticp 
- Ready to merge.

### CODE REVIEW

**Comment by sichvoge** [comment]
LGTM

**Comment by kabirbhartiRH** [comment]
I think we should specify the below as well to make it clearer in case of multiple apps with different labels.

_If you want to gather logs from apps with different labels to the same output, create multiple inputs with selectors for each type of app, and include them all as inputRefs to the same pipeline._

**Review by jc-berger — APPROVED** [review]
LGTM from doc point of view!

**Review by alanconway — APPROVED** [review]
All good!

**Review by jc-berger — APPROVED** [review]
New changes look great. LGTM from docs point of view.

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 1]
```diff
@@ -23,33 +25,47 @@ spec:
...
     ...
 ----
 <1> The name of the `ClusterLogForwarder` CR must be `instance`.
 <2> The namespace for the `ClusterLogForwarder` CR must be `openshift-logging`.
-<3> Specify the input for the pipeline.
-<4> Specify the output for the pipeline.
-<5> Specify the labels of pods whose log data you want to gather.
+<3> Specify the `name` value of one or more comma-separated `inputs` for the pipeline.
+<4> Specify the `name` value of one or more comma-separated `outputs` for the pipeline.
+<6> Define an input for each application that has a unique set of pod labels.
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:46]: This should be <5>

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 2]
```diff
@@ -23,33 +25,47 @@ spec:
   pipelines:
     - inputRefs: [ myAppLogData ] <3>
       outputRefs: [ default ] <4>
-  inputs:
+  inputs: <5>
     - name: myAppLogData
       application:
         selector:
           matchLabels:
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:32]: Add <6> adjacent to `matchLabels:`

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 3]
```diff
@@ -23,33 +25,47 @@ spec:
...
-            environment: production <5>
-            app: nginx <5>
-        namespaces: <6>
+            environment: production <6>
+            app: nginx <6>
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:34]: I am not sure <6> adjacent to `environment: production` and `app: nginx` is needed, based on the explanations to the call outs below. We probably need to confirm with the dev team.

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 4]
```diff
@@ -23,33 +25,47 @@ spec:
...
 ----
 <1> The name of the `ClusterLogForwarder` CR must be `instance`.
 <2> The namespace for the `ClusterLogForwarder` CR must be `openshift-logging`.
-<3> Specify the input for the pipeline.
-<4> Specify the output for the pipeline.
-<5> Specify the labels of pods whose log data you want to gather.
+<3> Specify the `name` value of one or more comma-separated `inputs` for the pipeline.
+<4> Specify the `name` value of one or more comma-separated `outputs` for the pipeline.
+<6> Define an input for each application that has a unique set of pod labels.
+<5> Use `matchLabels` to specify the labels of pods whose log data you want to gather. You must specify both a key and value, not just a key. To be selected, the pods must match all the key-value pairs.
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:47]: This should be <6>

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 5]
```diff
@@ -23,33 +25,47 @@ spec:
...
 <6> Optional: Specify one or more namespaces.
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:48]: This should be <7> on L 48

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 6]
```diff
@@ -23,33 +25,47 @@ spec:
...
 <7> Specify the output to forward your log data to. The optional `default` output shown here sends log data to the internal Elasticsearch instance.
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:49]: This should be <8> on L 49

File: modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc [hunk 7]
```diff
@@ -23,33 +25,47 @@ spec:
...
-        namespaces: <6>
+            environment: production <6>
+            app: nginx <6>
+        namespaces: <7>
         - app1
         - app2
-  outputs: <7>
-    - default <8>
+  outputs: <8>
+    - default <9>
```
Comment (Preeticp) [inline modules/cluster-logging-collector-log-forward-logs-from-application-pods.adoc:39]: Based on the explanation for the `outputs` call out below, the call out `<9>` is not required as the explanation for `outputs` covers the explanation for the `default` output too.

**Review by Preeticp — COMMENTED** [review]
Except for the call outs numbering part looks good to me @rolfedh 
Also, let me know if the merge needs to be held off till the end of May or if we can merge the PR as soon as the above issue is handled.

**Pull request merged**